src/Models/NonhydrostaticModels/update_nonhydrostatic_model_state.jl

```diff
@@ -10,6 +11,11 @@ import Oceananigans.TimeSteppers: update_state!
 Update peripheral aspects of the model (halo regions, diffusivities, hydrostatic pressure) to the current model state.
 """
 function update_state!(model::NonhydrostaticModel)
+    
+    # Mask immersed tracers
+    tracer_masking_events = mask_immersed_tracers!(model.tracers, model.architecture, model.grid)
```

Comment:
```suggestion
    tracer_masking_events = Tuple(mask_immersed_field!(c) for c in model.tracers)
```

Then we can get rid of `mask_immersed_tracers!`.